llama-index-core/llama_index/core/indices/knowledge_graph/base.py

```diff
@@ -240,6 +241,19 @@ def _insert(self, nodes: Sequence[BaseNode], **insert_kwargs: Any) -> None:
                     rel_embedding = self._embed_model.get_text_embedding(triplet_str)
                     self._index_struct.add_to_embedding_dict(triplet_str, rel_embedding)
 
+    def insert(self, document: Document, **insert_kwargs: Any) -> None:
```

Comment:
We have `insert_nodes` already that exists on the `BaseIndex` class. If we do implement this, then we should use `insert_documents` or `insert_nodes_from_documents` instead.

Reply:
@nerdai there's an insert() method on the base class already right? (Although it takes a single document, not a list, for whatever reason)

Reply:
ah yes! how did i miss that -- my bad. So in this case, this should already be available to KnowledgeGraphIndex